### PR TITLE
Add GET and PUT functionality to Debtors note

### PR DESCRIPTION
**Pull request by caismith-ch**
- Update debtors controller and service with the logic for GET and PUT functionality.
- Update tests to cover new additions

**Resolves**:  SFA-41 ,SFA-839

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/api/accounts/controller/DebtorsController.java [hunk 1]
```diff
@@ -83,4 +87,72 @@ public ResponseEntity create(@Valid @RequestBody Debtors debtors,
...
+
+        Transaction transaction = (Transaction) request.getAttribute(AttributeName.TRANSACTION.getValue());
+
+        ResponseEntity responseEntity;
+
+        try {
+            ResponseObject<Debtors> response = debtorsService
+                    .update(debtors, transaction, companyAccountId, request);
+
+            responseEntity = apiResponseMapper
```
Comment (ch-mcabrera) [inline src/main/java/uk/gov/companieshouse/api/accounts/controller/DebtorsController.java]: We could `return` here and in line `153` to be consistent with lines `128` and lines `133`. So then, we could get rid of the reference variable **ResponseEntity responseEntity;**

File: src/test/java/uk/gov/companieshouse/api/accounts/controller/DebtorsControllerTest.java [hunk 2]
```diff
@@ -57,6 +66,8 @@
 
     private static final String COMPANY_ACCOUNTS_ID = "companyAccountsId";
 
+    private static final String DEBTORS_ID = "debtorsId";
```
Comment (ch-mcabrera) [inline src/test/java/uk/gov/companieshouse/api/accounts/controller/DebtorsControllerTest.java]: I would move the constants to the top, `line 53`.

File: src/test/java/uk/gov/companieshouse/api/accounts/controller/DebtorsControllerTest.java [hunk 3]
```diff
@@ -27,6 +23,13 @@
 import uk.gov.companieshouse.api.accounts.utility.ApiResponseMapper;
 import uk.gov.companieshouse.api.accounts.utility.ErrorMapper;
 
+import javax.servlet.http.HttpServletRequest;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.*;
```
Comment (ch-mcabrera) [inline src/test/java/uk/gov/companieshouse/api/accounts/controller/DebtorsControllerTest.java]: Can we please add explicit imports?

File: src/test/java/uk/gov/companieshouse/api/accounts/controller/DebtorsControllerTest.java [hunk 4]
```diff
@@ -49,6 +52,12 @@
     @Mock
     DebtorsService debtorsService;
 
+    @Mock
+    private SmallFull smallFull;
```
[the remark this answers is not in the document]
Comment (caismith-ch) [reply src/test/java/uk/gov/companieshouse/api/accounts/controller/DebtorsControllerTest.java]: I agree, will prefix the mock variables.

File: src/test/java/uk/gov/companieshouse/api/accounts/controller/DebtorsControllerTest.java [hunk 5]
```diff
@@ -119,4 +130,134 @@ void createDebtorsBindingErrors() {
...
+    @Test
+    @DisplayName("Update debtors - no small full link")
+    void updateDebtorsNoSmallFullLink() {
+
+        when(request.getAttribute(AttributeName.SMALLFULL.getValue())).thenReturn(smallFull);
+        when(smallFull.getLinks()).thenReturn(smallFullLinks);
+        when(smallFullLinks.get(SmallFullLinkType.DEBTORS_NOTE.getLink())).thenReturn(null);
+
+        ResponseEntity responseEntity =
+                controller.update(debtors, bindingResult, COMPANY_ACCOUNTS_ID, request);
```
Comment (ch-mcabrera) [inline src/test/java/uk/gov/companieshouse/api/accounts/controller/DebtorsControllerTest.java]: Can we please follow the starndards regarding spaces, here and every file?
https://github.com/companieshouse/styleguides/blob/master/java.md#indentation

File: src/test/java/uk/gov/companieshouse/api/accounts/controller/DebtorsControllerTest.java (hunk 3 above)
Comment (caismith-ch) [reply src/test/java/uk/gov/companieshouse/api/accounts/controller/DebtorsControllerTest.java]: Yeah, my intellij seems to be doing it for static imports, will fix.

**Pull request merged**